gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformSingleNode.scala

```diff
@@ -303,7 +303,15 @@ object TransformOthers {
   class ReplaceSingleNode() extends LogLevelUtil with Logging {
 
     def doReplace(p: SparkPlan): SparkPlan = {
-      val plan = p
+      val plan = p match {
+        case plan: FileSourceScanExec
+            if plan.relation.fileFormat.getClass.getSimpleName == "ArrowCSVFileFormat" =>
+          val arrowScan = ArrowFileSourceScanExec(plan)
+          TransformHints.tagNotTransformable(arrowScan, "Arrow scan cannot transform")
+          return arrowScan
+        case p => p
+      }
+
       if (TransformHints.isNotTransformable(plan)) {
```

Comment:
shall we handle `ArrowFileSourceScanExec` inside this branch ?

Reply:
No, we the partitionFilters is null, it is taged not transformable, https://github.com/apache/incubator-gluten/blob/main/gluten-core/src/main/scala/org/apache/gluten/extension/columnar/TransformHintRule.scala#L343, otherwise, the plan has no tag

Reply:
The `ArrowFileSourceScanExec` is always tagged as fallback. So if we do not handle `ArrowFileSourceScanExec` here,  what happens if `ArrowFileSourceScanExec` has a dynamic partition filter ?

Reply:
The csv arrow reader does not support filter pushdown now, so all the filter will left to FilterExec